.github/workflows/test_openvino.yml

```diff
@@ -20,40 +18,58 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.12"]
+        test_pattern:
+          [
+            "*modeling.py",
+            "*modeling_diffusion.py",
+            "*quantization*",
+            "*training*",
+            "*export*",
+          ]
         transformers-version: ["4.36.0", "4.45.*"]
+        python-version: ["3.9"]
         os: [ubuntu-latest]
 
     runs-on: ${{ matrix.os }}
+
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
+
       - name: Setup Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          # install PyTorch CPU version to avoid installing CUDA packages on GitHub runner without GPU
+          pip install --upgrade pip
           pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu
-          pip install .[openvino,openvino-tokenizers,tests,diffusers] onnxruntime
-          pip install transformers==${{ matrix.transformers-version }}
+          pip install .[openvino,openvino-tokenizers,diffusers,tests] transformers[testing]==${{ matrix.transformers-version }}
+
+      - name: Assert versions
+        run: |
+          python -c "import transformers; print(transformers.__version__); assert transformers.__version__.startswith('${{ matrix.transformers-version }}'.replace('.*', ''))"
 
       - if: ${{ matrix.transformers-version == '4.36.0' }}
+        name: Downgrade Accelerate
         run: pip install accelerate==0.*
 
-      - name: Test with Pytest
+      - name: Test with Pytest (basic)
+        run: |
+          # catch failures early and quickly
+          pytest tests/openvino/test_modeling_basic.py --durations=0
+
+      - name: Test with Pytest (advanced)
+        run: |
+          pytest tests/openvino/${{ matrix.test_pattern }} --durations=0
         env:
           HF_HUB_READ_TOKEN: ${{ secrets.HF_HUB_READ_TOKEN }}
+
+      - name: Install dependencies (nightly)
         run: |
-          pytest tests/openvino/ --ignore tests/openvino/test_modeling_basic.py --durations=0
-      - name: Test basic
+          pip install --upgrade --pre openvino openvino-tokenizers --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
+
+      - name: Test with Pytest (nightly)
         run: |
-          pip uninstall -y nncf
```

Comment:
was added to make sure openvino models can be run even if nncf is not installed, since then nncf was added to the openvino extra but would prefer to keep it for now (as technically nncf shouldn't be a required dependency for openvino inference)

Reply:
made sure it's uninstalled for modeling tests